### DISCUSSION TITLE
Detect .NET 7 with WIX V3.11

### DISCUSSION THREAD

**Discussion by AlquistArj** [wixtoolset/issues#7172] [Questions]
What's the best way to check if .NET 7 is installed using WIX V3.11? I know it's a new feature in V4 but my installer is V3.11 based.

**Comment by robmen** [comment]
The best way? IMHO, upgrade to WiX v4. 😄

**Reply by AlquistArj** [reply]
I've upgraded one of my projects to WiX v4 but I'm still confused how to detect if .NET 7 Desktop is installed. I'm trying to keep it simple so I'm hoping to exit with a warning message if .NET 7 Desktop is not yet installed.

I'm guessing I need to use netfx:DotNetCoreSearch but I cannot work out where to place it without causing errors.

**Reply by barnson** [reply]
In a package, use `DotNetCompatibilityCheck`.

**Reply by barnson** [reply]
https://wixtoolset.org/docs/tools/wixext/dotnet/#detecting-net-in-msi-packages

**Reply by AlquistArj** [reply]
So at the top of my .wsx file I have 

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:ui="http://wixtoolset.org/schemas/v4/wxs/ui" xmlns:netfx="http://wixtoolset.org/schemas/v4/wxs/netfx">
	<Package Name="CopyDDFFolders" Manufacturer="Me" Language="1033" Version="1.22.0.0" 
			 UpgradeCode="04CD4A13-1B2B-442C-8FDF-E0DA44E49A4A" 
			 InstallerVersion="301" 
			 ProductCode="60993416-FACA-4628-9CF2-357C7F89AB51">
		<netfx:DotNetCompatibilityCheck
			  Property="DOTNETRUNTIMECHECK"
			  RollForward="major"
			  RuntimeType="desktop"
			  Version="8.0.0"
			  Platform="x64" />
```

I'm using 8.0.0 here (which isn't installed on my machine) so I can test it on my machine. It doesn't seem to prevent the installer from running. Do I need some sort of condition check? Does that go inside the <netfx:DotNetCompatibilityCheck></netfx:DotNetCompatibilityCheck?

.ps the example in **Detecting .NET in MSI packages** doesn't work. It only wants a three digit version and needs a Platform attribute.

**Reply by barnson** [reply]
`DotNetCompatibilityCheck` only sets the property.

**Reply by AlquistArj** [reply]
Ok, so how do I stop the installation and display a message based on the property value? In WIX 3 I used to have something like this but with V4 I get an "Invalid child element" error.
`		<netfx:DotNetCompatibilityCheck
			  Property="DOTNET8RUNTIMECHECK"
			  RollForward="major"
			  RuntimeType="desktop"
			  Version="8.0.0"
			  Platform="x64">
		</netfx:DotNetCompatibilityCheck>

		<Condition Message="This application requires .NET Core Desktop V8.">
			<![CDATA[DOTNET8RUNTIMECHECK]]>
		</Condition>
		`

**Reply by BMurri** [reply]
Further up in the same page shows an example of using the `Launch` element with the `Condition` attribute to perform the same concept.

**Reply by AlquistArj** [reply]
Ok, so I now have this but it doesn't stop the installer from running (I don't have .NET 8). I'm guessing I have something wrong in Condition but I'm struggling to find the right format for conditions in the V4 docs.

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:ui="http://wixtoolset.org/schemas/v4/wxs/ui" xmlns:netfx="http://wixtoolset.org/schemas/v4/wxs/netfx">
	<Package Name="CopyDDFFolders" Manufacturer="Me" Language="1033" Version="1.22.0.0" 
			 UpgradeCode="04CD4A13-1B2B-442C-8FDF-E0DA44E49A4A" 
			 InstallerVersion="301" 
			 ProductCode="60993416-FACA-4628-9CF2-357C7F89AB51">
		<netfx:DotNetCompatibilityCheck
			  Property="DOTNET8RUNTIMECHECK"
			  RollForward="major"
			  RuntimeType="desktop"
			  Version="8.0.0"
			  Platform="x64">
		</netfx:DotNetCompatibilityCheck>

		<Launch
		  Message="This application requires .NET Core Desktop V8 to be installed."
		  Condition="DOTNET8RUNTIMECHECK"	/>
```

**Reply by BMurri** [reply]
Logs?

**Comment by rseanhall** [comment]
https://stackoverflow.com/questions/67024261/how-to-install-a-net-core-application-with-net-core-runtime-prerequisite-from-w/67198122#67198122
